### DISCUSSION TITLE
Points with spheres?

### DISCUSSION THREAD

**Discussion by bknill** [pmndrs/drei#958] [Q&A]
I'm trying to change the geometry of points. Currently they're squares of [1,1]

This doesn't work. 

      <Points positions={postionsBufffer} sizes={sizesBuffer}>
         <sphereGeometry attach={'geometry'} args={[1, 64, 64]} />
        <pointsMaterial />
      </Points>

Is there a way it can be done?

**Comment by drcmda** [comment]
points doesnt have a geometry, attaching one will do nothing. three.pointsmaterial by default is a square, you can alter the shape with a shader material. drei actually has an anti aleased sphere point shader: https://github.com/pmndrs/drei#pointmaterial

it is based on regular pointsmaterial and overrides the shader, otherwise it supports all the regular props. you can use it in `<Points>` as well as in three.points.

https://codesandbox.io/s/r3f-points-forked-eq7sc?file=/src/App.js
